alphagov/paas-cf
# [#119269637] Move to cloud config

## What

[Move to cloud config](https://www.pivotaltracker.com/n/projects/1275640/stories/119269637). This PR includes:
* new manifest directory structure
* new, unified way to render manifests
* cleanup of unnecessary properties and settings
* various minor fixes to make move easier
* move of cf-manifest to cloud config + manifest
* update pipeline to facilitate cloud config rendering and stemcell upload (as this is now not done by BOSH anymore)
* further simplify manifest where possible
* adjust tests: 
  -  remove tests for things that are no longer in cf-manifest
  -  add tests for cloud-config
  -  add cloud-config and cf-manifest integration tests
* ensure persistent data and job names are migrated
* ensure interrupt-less future deploys

# NOTE: THERE'S NO WAY BACK AFTER CLOUD CONFIG
if you want to go back to resource_pools and manifest only, you need to delete your BOSH (inc. database). Don't forget to run cleanup [repeatedly](https://github.com/cloudfoundry/bosh/issues/1294)
## How to review

Do at least these 3 intependent tests:

* Review commits and code
* Build from scratch. All tests should pass.
* Apply against existing deployment (built from master). Upgrade/migration takes about 1 hour, there is some downtime. Verify that persistent data for graphite and elasticsearch was migrated correctly. You should be able to see historical data in the deployment after migration. All tests should pass.

## Who can review

Not @mtekel or @HenryTK or @saliceti
